### PR TITLE
Bundler: allows configuring the output path.

### PR DESCRIPTION
**Pull request by pomek**
Fixes: #4, #8 

I have to re-open the same issue in the main repository because this PR changes API of exported module.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -29,15 +29,29 @@ const utils = require( './utils' );
 module.exports = ( config ) => {
 	const sourceBuildDir = path.join( config.ROOT_DIR, config.BUILD_DIR, 'esnext' );
 	const bundleDir = path.join( config.ROOT_DIR, config.BUNDLE_DIR );
-	const bundleTmpDir = path.join( bundleDir, 'tmp' );
-	const temporaryEntryFilePath = path.join( bundleTmpDir, 'entryfile.js' );
 
 	const tasks = {
 		/**
 		 * Removes all files from bundle directory.
+		 *
```
Comment (oskarwrobel) [inline lib/index.js]: Missing @param doc.

File: lib/index.js [hunk 2]
```diff
@@ -49,15 +63,19 @@ module.exports = ( config ) => {
 		 * @returns {Promise} Promise that resolve bundling for CSS and JS.
 		 */
 		generate( config ) {
+			mkdirp.sync( bundleDir );
```
Comment (oskarwrobel) [inline lib/index.js]: This line should be under the comment `Create a temporary entry file with proper directory structure if not exist.`

File: lib/index.js [hunk 3]
```diff
@@ -49,15 +63,19 @@ module.exports = ( config ) => {
 		 * @returns {Promise} Promise that resolve bundling for CSS and JS.
```
Comment (oskarwrobel) [inline lib/index.js]: We can add more specific doc for config param, including all available options with default values.

File: lib/index.js [hunk 4]
```diff
@@ -138,56 +162,82 @@ module.exports = ( config ) => {
...
+			cssFromConfig() {
+				try {
+					return tasks.minify.css( loadBundleConfig() );
+				} catch ( err ) {
+					return rejectProcess( err );
+				}
+			},
+
 			/**
 			 * JS minification by UglifyJS.
```
Comment (oskarwrobel) [inline lib/index.js]: missing `@param`

File: lib/index.js [hunk 5]
```diff
@@ -138,56 +162,82 @@ module.exports = ( config ) => {
...
+
+				let stream = gulp.src( `${ filePath }.js` )
 					.pipe( gulpUglify() );
 
-				return utils.saveFileFromStreamAsMinified( stream, bundleDir );
+				return utils.saveFileFromStreamAsMinified( stream, path.dirname( filePath ) );
 			},
 
 			/**
 			 * CSS minification by cssnano.
```
Comment (oskarwrobel) [inline lib/index.js]: `@param`

**Comment by pomek** [comment]
> Maybe it would be better to load configuration from file once in this place where we registers gulp tasks, and use config object? 

Yesterday I tried to do this but Gulp threw an error even config file isn't needed. This mechanism has to be a little clever and knows when config is necessary and when is unnecessary. 

> But now there is few new, small function you can test.

Sure but we don't still know whether this API is final API. Docs-builder is still under heavy development and everything can change.

Thanks for review. I will add missing things tomorrow.

**Pull request merged**